spaces/importers/units.py

```diff
@@ -131,4 +146,63 @@ def create_unit(self, importer_data: dict) -> bool:
             unit=unit, defaults=location_data
         )
 
+        self.imported_unit_ids.append(unit.id)
+
         return unit_created
+
+
+class UnitHaukiResourceIdImporter:
+    @transaction.atomic
+    def import_hauki_resource_ids_for_units(
+        self,
+        unit_ids: Optional[List[int]] = None,
+        tprek_ids: Optional[List[str]] = None,
+    ):
+        if not unit_ids and not tprek_ids:
+            raise ValueError("Either unit_ids or tprek_ids is required.")
+
+        if unit_ids:
+            units = Unit.objects.filter(id__in=unit_ids)
+        else:
+            units = Unit.objects.filter(tprek_id__in=tprek_ids)
+
+        url = settings.HAUKI_API_URL
+        url = urljoin(url, "/v1/resource/")
+
+        logger.info(f"Importing units {units} resource ids from url {url}")
+
+        params = {
+            "data_source": "tprek",
+            "origin_id_exists": True,
+            "page_size": 50000,
+        }
+
+        self.resource_id_map = {}
+        counter = 0
+
+        def read_response(counter):
+            counter += 1
+            logger.info(f"Fetching from hauki. Page number: {counter}")
+            for resource in data["results"]:
+                for origin in resource["origins"]:
+                    if origin["data_source"]["id"] == "tprek":
+                        self.resource_id_map[origin["origin_id"]] = resource["id"]
+            return counter
+
+        data = make_hauki_get_request(url, params)
+        counter = read_response(counter)
+        next = data.get("next", False)
+        while next:
+            data = make_hauki_get_request(next, params)
+            counter = read_response(counter)
+            next = data.get("next", False)
```

Comment:
I think this could be simplified to:
```python
next = url
while next:
    data = make_hauki_get_request(next, params)
    counter = read_response(counter)
    next = data.get("next", False)
```

Reply:
This whole paging was a bummer since the `page_size` is limited to 1000... Hoping we could get rid of it in the future